Sources/LiveKit/Support/DeviceManager.swift

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024 LiveKit
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import AVFoundation
+
+// Internal-only for now
+class DeviceManager: Loggable {
+    // MARK: - Public
+
+    public static let shared = DeviceManager()
+
+    public static func prepare() {
+        // Instantiate shared instance
+        _ = shared
+    }
+
+    public var devices: [AVCaptureDevice] { _state.devices }
+    public let devicesCompleter = AsyncCompleter<[AVCaptureDevice]>(label: "devices", defaultTimeout: 10)
+
+    private lazy var discoverySession: AVCaptureDevice.DiscoverySession = {
+        let deviceTypes: [AVCaptureDevice.DeviceType]
+        #if os(iOS)
+        deviceTypes = [
+            .builtInDualCamera,
```

Comment:
would these show up as diff devices? or the same? if they are all different, it would be confusing right?

Reply:
I think the device manager should bring aware of all devices but for the user facing device enumeration API, i can add a filter option (default on)